lumicks/pylake/nb_widgets/range_selector.py

```diff
@@ -3,34 +3,53 @@
 import time
 
 
-class SliceRangeSelector:
-    def __init__(self, channel_slice, axes=None, show=True):
+class BaseRangeSelectorWidget:
+    """ Base class for range selection widgets.
+        Takes care of setting up connections and internal event handling.
+
+        Parameters
+        __________
+        axes : matplotlib.axes.Axes
+            Axes instance used for plotting
+        show : bool
+            Controls if plot is rendered immediately
+        range_conversion_function : function
+            Function to convert from units of the underlying data to plotting units,
+            for example: timestamps -> seconds in the case of a Slice
+            default simply returns the value unchanged.
+    """
+    def __init__(self, axes=None, show=True, range_conversion_fcn=lambda point:point):
         self._axes = axes if axes else plt.axes(label=f"lk_slice_widget_{time.time()}")
-        self.slice = channel_slice
-        self.time = self.slice.timestamps
         self.connection_id = None
         self.current_range = []
         self._ranges = []
+        self.range_conversion_fcn = range_conversion_fcn
         self.open = show
 
         if show:
             self._axes.get_figure().canvas.mpl_connect('close_event', self.close)
             self.update_plot()
 
+    @property
+    def xdata(self):
+        raise NotImplementedError
```

Comment:
Public?

Reply:
👍 wasn't sure if it was necessary to make it internal since these classes are internal anyway, but better safe than sorry